network.cpp

```diff
@@ -255,33 +255,35 @@ void test_network() //!OCLINT
 
 //    }
 
-//    ///Network weights mutate following a normal distribution
-//    {
-//        double mut_rate = 0.01;
-//        double mut_step = 0.1;
-//        std::mt19937_64 rng;
+    ///Network weights mutate following a normal distribution
+    {
+        double mut_rate = 0.01;
+        double mut_step = 0.1;
+        std::mt19937_64 rng;
 
-//        auto expected_mean_value  = 0;
-//        auto expected_stdev = mut_step;
+        auto expected_mean_value  = 0;
+        auto expected_stdev = mut_step;
 
-//        int repeats = 100000;
+        int repeats = 100000;
 
-//        auto very_simple_nodes = std::vector<int>{1,2,1};
-//        network n{very_simple_nodes};
+        auto very_simple_nodes = std::vector<int>{1,2,1};
+        network n{very_simple_nodes};
 
-//        std::vector<double> networks_weights = register_n_mutations(n,
-//                                                                    mut_rate,
-//                                                                    mut_step,
-//                                                                    rng,
-//                                                                    repeats);
+        std::vector<weight> networks_weights = register_n_mutations(n,
+                                                                    mut_rate,
+                                                                    mut_step,
+                                                                    rng,
+                                                                    repeats);
 
-//        auto mean = calc_mean(networks_weights);
-//        double stdev = calc_stdev(networks_weights);
+        std::vector<double> weights_as_doubles = convert_to_double(networks_weights);
 
-//        assert(mean - expected_mean_value < 0.01 && mean - expected_mean_value > -0.01);
-//        assert(stdev - expected_stdev < 0.01 && stdev - expected_stdev > -0.01);
+        auto mean = calc_mean(weights_as_doubles);
+        double stdev = calc_stdev(weights_as_doubles);
```

Comment:
maybe kee the `type` writing consistent. In this case I think `double` is better
```suggestion
        double mean = calc_mean(weights_as_doubles);
        double stdev = calc_stdev(weights_as_doubles);
```